WordPress/gutenberg
# Remove specificify for link color

See https://github.com/WordPress/gutenberg/pull/30541#issuecomment-832553715

- #29557 introduced the `:not(.wp-block-button_link)` negation so buttons within groups didn't get styled with the link color
- #30541 ported that changed to global styles.

It was reported to cause issues with navigations but also with any other block that may use the `a` element. This PR explores removing that negation.

## How to test

- Add a group block with a paragraph containing a link plus a button within.
- Set the background and link colors of the group.
- Verify that the link within the paragraph gets the link color but the button text doesn't.
